src/main/java/blusunrize/immersiveengineering/common/blocks/metal/SawmillTileEntity.java

```diff
@@ -0,0 +1,596 @@
+			list.add(new AxisAlignedBB(.25, 0, .125, .875, .875, .75));
+			list.add(new AxisAlignedBB(.1875, 0, 0, .9375, .125, .8125));
+			return list;
+		}
+		if(new BlockPos(1, 0, 2).equals(posInMultiblock))
+		{
+			List<AxisAlignedBB> list = Lists.newArrayList(new AxisAlignedBB(0, 0, 0, 1, .5, 1));
+			list.add(new AxisAlignedBB(.1875, .5, 0, .9375, 1, .8125));
+			list.add(new AxisAlignedBB(.9375, .5, .25, 1, .875, .625));
+			return list;
+		}
+		if(new BlockPos(2, 0, 2).equals(posInMultiblock))
+		{
+			List<AxisAlignedBB> list = Lists.newArrayList(new AxisAlignedBB(0, 0, 0, 1, .5, 1));
+			list.add(new AxisAlignedBB(0, .5, .25, 1, .875, .625));
+			return list;
+		}
+		// Conveyors
+		if(posInMultiblock.getY()==1&&posInMultiblock.getZ()==1)
+			return VoxelShapes.create(0, 0, 0, 1, .125, 1).toBoundingBoxList();
+		// Rest
+		return VoxelShapes.create(0, 0, 0, 1, 1, 1).toBoundingBoxList();
+	}
+
+	@Override
+	public Set<BlockPos> getEnergyPos()
+	{
+		return ImmutableSet.of(
+				new BlockPos(2, 1, 0)
+		);
+	}
+
+	@Override
+	public Set<BlockPos> getRedstonePos()
+	{
+		return ImmutableSet.of(
+				new BlockPos(0, 1, 2)
+		);
+	}
+
+	@Override
+	public boolean shouldRenderAsActive()
+	{
+		return getEnergyStored(null) > 0&&!isRSDisabled()&&!this.sawblade.isEmpty();
+	}
+
+	@Override
+	public void replaceStructureBlock(BlockPos pos, BlockState state, ItemStack stack, int h, int l, int w)
+	{
+		super.replaceStructureBlock(pos, state, stack, h, l, w);
+		if(h==1&&l==1)
+		{
+			TileEntity tile = world.getTileEntity(pos);
+			if(tile instanceof ConveyorBeltTileEntity)
+				((ConveyorBeltTileEntity)tile).setFacing(this.getIsMirrored()?this.getFacing().rotateYCCW(): this.getFacing().rotateY());
+		}
+	}
+
+	private void insertItemToProcess(ItemStack stack, boolean simulate)
+	{
+		if(this.sawmillProcessQueue.size() < this.getProcessQueueMaxLength())
+		{
+			float dist = 1;
+			SawmillProcess p = null;
+			if(this.sawmillProcessQueue.size() > 0)
+			{
+				p = this.sawmillProcessQueue.get(this.sawmillProcessQueue.size()-1);
+				if(p!=null)
+					dist = p.getRelativeProcessStep();
+			}
+			if(p!=null&&dist < this.getMinProcessDistance(null))
+				return;
+			if(!simulate)
+			{
+				p = new SawmillProcess(Utils.copyStackWithAmount(stack, 1));
+				this.sawmillProcessQueue.add(p);
+				this.markDirty();
+				this.markContainingBlockForUpdate(null);
+			}
+			stack.shrink(1);
+		}
+	}
+
+	@Override
+	public void onEntityCollision(World world, Entity entity)
+	{
+		if(new BlockPos(0, 1, 1).equals(posInMultiblock)&&!world.isRemote&&entity!=null&&entity.isAlive()&&entity instanceof ItemEntity)
+		{
+			SawmillTileEntity master = master();
+			if(master==null)
+				return;
+			ItemStack stack = ((ItemEntity)entity).getItem();
+			if(stack.isEmpty())
+				return;
+			master.insertItemToProcess(stack, false);
+			if(stack.getCount() <= 0)
+				entity.remove();
+		}
```

Comment:
I'm disappointed, stepping on the giant exposed spinning sawblade does not hurt you?

Reply:
...you make a good point.